src/overlays/actors/ovl_En_Niw/z_en_niw.c

```diff
@@ -21,61 +42,1136 @@ const ActorInit En_Niw_InitVars = {
+    }
+    if (this->unk_2D0 != this->unk_26C[1]) {
+        Math_SmoothScaleMaxF(&this->unk_2D0, this->unk_26C[1], 0.8f, 7000.0f);
+    }
+    if (this->unk_2D4 != this->unk_26C[5]) {
+        Math_SmoothScaleMaxF(&this->unk_2D4, this->unk_26C[5], 0.8f, 7000.0f);
+    }
+    if (this->unk_2D8 != this->unk_26C[6]) {
+        Math_SmoothScaleMaxF(&this->unk_2D8, this->unk_26C[6], 0.8f, 7000.0f);
+    }
+}
+
+void EnNiw_SpawnAttackCucco(EnNiw* this, GlobalContext* globalCtx) {
+    f32 viewX;
+    f32 viewY;
+    f32 viewZ;
+    Vec3f attackCuccoPos;
+    EnAttackNiw* attackCucco;
+
+    if (this->timer5 == 0) {
+        if (this->unk_296 < 7) {
+            viewX = globalCtx->view.lookAt.x - globalCtx->view.eye.x;
+            viewY = globalCtx->view.lookAt.y - globalCtx->view.eye.y;
+            viewZ = globalCtx->view.lookAt.z - globalCtx->view.eye.z;
+            attackCuccoPos.x = ((Math_Rand_ZeroOne() - 0.5f) * viewX) + globalCtx->view.eye.x;
+            attackCuccoPos.y = Math_Rand_CenteredFloat(0.3f) + ((globalCtx->view.eye.y + 50.0f) + (viewY * 0.5f));
+            attackCuccoPos.z = ((Math_Rand_ZeroOne() - 0.5f) * viewZ) + globalCtx->view.eye.z;
+            attackCucco = Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_EN_ATTACK_NIW,
+                                             attackCuccoPos.x, attackCuccoPos.y, attackCuccoPos.z, 0, 0, 0, 0);
+
+            if (attackCucco != NULL) {
+                this->unk_296++;
+                this->timer5 = 10;
+            } else {
+                osSyncPrintf("\n\n");
+                osSyncPrintf(VT_FGCOL(GREEN) " ☆☆☆☆☆ 発生できず  ☆☆☆☆☆ \n" VT_RST);
+            }
+        }
+    }
+}
+
+void func_80AB6100(EnNiw* this, GlobalContext* globalCtx, s32 arg2) {
+    f32 factor;
+    f32 targetRotY;
+
+    if (this->timer4 == 0) {
+        this->timer4 = 3;
+
+        if (this->actor.bgCheckFlags & 1) {
+            this->actor.velocity.y = 3.5f;
+        }
+    }
+    if (this->timer3 == 0) {
+        this->unk_2A0++;
+        this->unk_2A0 &= 1;
+        this->timer3 = 5;
+    }
+    if (this->unk_2A0 == 0) {
+        factor = D_80AB860C[arg2];
+    } else {
+        factor = -D_80AB860C[arg2];
+    }
+    if (arg2 == 1) {
+        if (this->timer6 == 0 || this->actor.bgCheckFlags & 8) {
+            this->timer6 = 150;
+            if (this->timer8 == 0) {
+                this->timer8 = 70;
+                this->unk_2E4 = this->actor.yawTowardsLink;
+            }
+        }
+    }
+    targetRotY = this->unk_2E4 + factor;
+    Math_SmoothScaleMaxMinS(&this->actor.posRot.rot.y, targetRotY, 3, this->unk_2FC, 0);
+    Math_SmoothScaleMaxF(&this->unk_2FC, 3000.0f, 1.0f, 500.0f);
+    func_80AB5BF8(this, globalCtx, 5);
+}
+
+void EnNiw_ResetAction(EnNiw* this, GlobalContext* globalCtx) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_060000E8, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_060000E8), 0, -10.0f);
+
+    switch (this->actor.params) {
+        case 0x4:
```

Comment:
same thing with decimal. Gonna stop commenting it, but I think all of these actor param checks (even the ones not in cases) should be decimal